tools/dds/dds-server/rs-dds-server.cpp

```diff
@@ -96,114 +96,172 @@ void add_init_device_header_msg( rs2::device dev, std::shared_ptr<librealsense::
     }
+    motion_stream_profiles_msg.num_of_profiles = index;
 }
 
 void add_init_profiles_msgs( rs2::device dev, std::shared_ptr<librealsense::dds::dds_device_server> &server )
 {
     using namespace librealsense::dds::topics;
-    auto stream_idx = 0;
-    device::notifications::video_stream_profiles video_stream_profiles_msg;
-    device::notifications::motion_stream_profiles motion_stream_profiles_msg;
+    auto sensor_idx = 0;
 
     // For each sensor publish all it's profiles
     for( auto &sensor : dev.query_sensors() )
     {
+        device::notification::sensor_header_msg::sensor_type sensor_type;
+
+        if( sensor.is< rs2::color_sensor >() || sensor.is< rs2::depth_sensor >() )
+            sensor_type = device::notification::sensor_header_msg::VIDEO_SENSOR;
+        else if( sensor.is< rs2::motion_sensor >() )
+            sensor_type = device::notification::sensor_header_msg::MOTION_SENSOR;
+        else
+            throw std::runtime_error(
+                "Sensor type is not supported (only video & motion sensors are supported)" );
+
         auto && stream_profiles = sensor.get_stream_profiles();
-        send_stream_header_msg( server, sensor, stream_idx++, stream_profiles.size() );
-
-        // Prepare stream profiles messages
-        prepare_profiles_messeges( dev,
-                                   stream_profiles,
-                                   video_stream_profiles_msg,
-                                   motion_stream_profiles_msg );
-
-        // Send video stream profiles
-        raw::device::notifications raw_video_stream_profiles_msg;
-        device::notifications::construct_raw_message(
-            device::notifications::msg_type::VIDEO_STREAM_PROFILES,
-            video_stream_profiles_msg,
-            raw_video_stream_profiles_msg );
-        server->add_init_msgs( raw_video_stream_profiles_msg );
-
-        // Send motion stream profiles
-        raw::device::notifications raw_motion_stream_profiles_msg;
-        device::notifications::construct_raw_message(
-            device::notifications::msg_type::MOTION_STREAM_PROFILES,
-            motion_stream_profiles_msg,
-            raw_motion_stream_profiles_msg );
-        server->add_init_msgs( raw_motion_stream_profiles_msg );
-
-        // Send pose stream profiles ? TODO
+        send_sensor_header_msg( server, sensor, sensor_type, sensor_idx );
+
+        switch( sensor_type )
+        {
+        case device::notification::sensor_header_msg::VIDEO_SENSOR: 
+            {
+                device::notification::video_stream_profiles_msg video_stream_profiles_msg;
+                prepare_video_profiles_messeges( dev,
+                                                 sensor_idx,
+                                                 stream_profiles,
+                                                 video_stream_profiles_msg );
+
+                // Send video stream profiles
+                if( video_stream_profiles_msg.num_of_profiles > 0 )
```

Comment:
If it's 0 then we don't send anything? The device will wait and time out...

Reply:
A sensor with no profiles is not a use case we should address IMO.
So the device will wait and time out..

Reply:
You made it a use-case when you added the if :)